internal/impl/google/drive_search.go

```diff
@@ -0,0 +1,158 @@
+			service.NewStringListField(driveSearchFieldProjection).
+				Description("The partial fields to include in the result.").
+				Default([]any{"id", "name", "mimeType", "parents", "size"}),
+			service.NewIntField(driveSearchFieldMaxResults).
+				Description("The maximum number of results to return.").
+				Default(64),
+		).
+		Example("Search & download files from Google Drive", "This examples downloads all the files from Google Drive that are returned in the query", `
+input:
+  stdin: {}
+pipeline:
+  processors:
+    - google_drive_search:
+        query: "${!content().string()}"
+        projection: ["id", "name", "mimeType", "parents", "size"]
+    - mutation: 'meta path = this.name'
+    - google_drive_download:
+        file_id: "${!this.id}"
+        mime_type: "${!this.mimeType}"
+output:
+  file:
+    path: "${!@path}"
+    codec: all-bytes
+`)
+}
+
+type googleDriveSearchProcessor struct {
+	*baseProcessor
+	query      *service.InterpolatedString
+	fields     []string
+	maxResults int
+}
+
+// newGoogleDriveSearchProcessor creates a new instance of googleDriveSearchProcessor.
+func newGoogleDriveSearchProcessor(conf *service.ParsedConfig, mgr *service.Resources) (service.Processor, error) {
+	base, err := newBaseProcessor(conf)
+	if err != nil {
+		return nil, err
+	}
+	query, err := conf.FieldInterpolatedString(driveSearchFieldQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	fields, err := conf.FieldStringList(driveSearchFieldProjection)
+	if err != nil {
+		return nil, err
+	}
+
+	maxResults, err := conf.FieldInt(driveSearchFieldMaxResults)
+	if err != nil {
+		return nil, err
+	}
+
+	return &googleDriveSearchProcessor{
+		baseProcessor: base,
+		query:         query,
+		fields:        fields,
+		maxResults:    maxResults,
+	}, nil
+}
+
+var errStopIteration = errors.New("stop iteration")
+
+func (g *googleDriveSearchProcessor) Process(ctx context.Context, msg *service.Message) (service.MessageBatch, error) {
+	client, err := g.getDriveService(ctx)
+	if err != nil {
+		return nil, err
+	}
+	q, err := g.query.TryString(msg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to interpolate query: %v", err)
+	}
+	call := client.Files.List().
+		Context(ctx).
+		Q(q).
+		PageSize(min(int64(g.maxResults), 100)).
+		Fields("nextPageToken", googleapi.Field("files("+strings.Join(g.fields, ",")+")"))
+	var files []*drive.File
+	err = call.Pages(ctx, func(page *drive.FileList) error {
+		files = append(files, page.Files...)
+		if len(files) >= g.maxResults {
+			return errStopIteration
+		}
+		return nil
+	})
+	if errors.Is(err, errStopIteration) {
+		err = nil
+	}
+	if err != nil {
+		return nil, fmt.Errorf("failed to query files in google drive: %v", err)
+	}
+	batch := service.MessageBatch{}
+	for _, file := range files {
+		b, err := file.MarshalJSON()
+		if err != nil {
+			return nil, fmt.Errorf("failed to marshal file to JSON: %v", err)
+		}
+		msg := service.NewMessage(b)
```

Comment:
Thinking to maybe copy the current message so we preserve existing metadata fields

Reply:
Done!